vivisect/parsers/elf.py

```diff
@@ -613,36 +621,56 @@ def loadElfIntoWorkspace(vw, elf, filename=None, baseaddr=None):
             sva += baseoff
             if symname:
                 vw.makeName(sva, symname, filelocal=True, makeuniq=True)
-                valu = vw.readMemoryPtr(sva)
-                if not vw.isValidPointer(valu) and s.st_size == vw.psize:
-                    vw.makePointer(sva, follow=False)
+                if sva == 0x0:
+                    # if the address of the object is 0x0,
+                    # then it is probably an extern object, like the environ pointer provided by glibc.
+                    pass
+                elif not vw.isValidPointer(sva):
```

Comment:
Are these checks redundant assuming that `0x0` is not a valid pointer in an Elf binary?

Reply:
My understanding is that the compiler will emit 0x0 for extern objects, so I wanted an explicit case and comments for that. The second case is just to more thoroughly handle "what if we find junk data", which I did see in another sample.

Reply:
is it beneficial to log a debug message here as well?